### ISSUE TITLE
Add  Skrill - Multi-currency payment method

### ISSUE THREAD

**Issue by pazza83** [bisq-network/growth#232]
## Skrill - Multi-currency

Website: https://www.skrill.com/
Fees: https://www.skrill.com/en/siteinformation/fees/#send-money-receive-money
Terms and conditions: https://www.skrill.com/en/footer/terms-conditions/skrillaccounttermsofuse/
Skrill Wiki: https://en.wikipedia.org/wiki/Skrill

### Background
Multi-currency payment methods are a good addition to Bisq as they increase accessibility to markets for more users.

### Why
Skrill is an established multi-currency payment service.

### Region
Multiple counties over 120+ 

Here is a list of all the countries that are [not serviced by Skrill](https://www.skrill.com/en/footer/terms-conditions/non-serviced-countries/)

### Currencies
Multiple currencies including:

- AUD
- BDT
- BRL
- EUR
- GBP
- IDR
- INR
- KES
- LKR
- MXN
- MYR
- NPR
- PHP
- PKP
- PLN
- THB
- USD
- VDN

### Chargeback risk
Low

### Size of user base

Skrill has over 40 million users worldwide.

### Duration
From 1-3 business days. Therefore, recommend 4 days as trade time.

### Fees

Free for international transfers (using independent exchange rate provided by Reuters)

2% fee for sending money domestically.

Users will be aware what it costs to send.

### Fraud risk
Low

### Payment amounts
Within 0.25 BTC limit

### Additional information

Payments can be funded by credit and debit card.
 
I originally discounted this payment method as no to Skrill to Skrill payment is possible, but expensive (1% to deposit funds, 1,5% to send funds, and 1% to withdraw funds). 

However if technically possible my proposal is for Skrill to be a 'taker' account only. Similar to my proposal for [WorldRemit](https://github.com/bisq-network/growth/issues/230)

Skrill accounts would be able to take offers on any of the following payment methods:

SEPA (EUR)
Faster Payments (GBP)
Wire Transfer US (USD)
National Bank Transfer (in any of the currencies above)

**Comment by RififiCastorjunior** [comment]
A foot note say this : 

> Important notice

>As of 22 February 2021 the following Fees shall apply:
Withdrawal Fee from a Skrill Account to a NETELLER Account – 3.49% per transaction
International transfer send money Fee for the use of Skrill Money Transfer – up to 4.99% per transaction - applicable to international money transfers in the same send and receive currency only
International transfer send money FX markup Fee - up to 4.99% per transaction.

Almost 5% of all transfer

**Comment by pazza83** [comment]
> A foot note say this :
> 
> > Important notice
> 
> > As of 22 February 2021 the following Fees shall apply:
> > Withdrawal Fee from a Skrill Account to a NETELLER Account – 3.49% per transaction
> > International transfer send money Fee for the use of Skrill Money Transfer – up to 4.99% per transaction - applicable to international money transfers in the same send and receive currency only
> > International transfer send money FX markup Fee - up to 4.99% per transaction.
> 
> Almost 5% of all transfer

Yes the high fees for Skrill to Skrill transactions put me off at first. However, the way I have proposed to implement it means that these fees will not be payable. 

Say if I was selling BTC for EUR, you would be able to send me any other currency from the list above to pay. The only cost would be the difference between the mid market exchange rate and what you were charged, Your fees would be very low likely 0.25-0.5%.

I would receive all the Euros in my bank and then could transfer them out for free.

**Comment by Conza88** [comment]
> I would receive all the Euros in my bank and then could transfer them out for free.

Yep, that'd be the way. No Skill to skill, but however best. For that low level fee's as taker only.

**Comment by pazza83** [comment]
@bisq-network/dev is a 'taker only' payment method something that can be achieved?

**Comment by sqrrm** [comment]
How would this work? User sets up a skrill account, won't be able to create offers but it would match to sepa/faster payment/wire/national bank options while looking to take offers? Does the information for all those methods fit with the skrill side of things or is something extra needed?

Sounds doable to have a take only account type, but is that really what you want to achieve here? Isn't it buy only since the skrill user will be sending fiat?

**Comment by Conza88** [comment]
> Isn't it buy only since the skrill user will be sending fiat?

Yeah 'buyer of BTC' only seems more accurate, whether taking offers for BTC or making offers for BTC.

I'd be real keen for the universal fiat nature of it, in terms of opening up liquidity to all fiat markets essentially to take offers. 

Is it just opening access to national/local banks? Or would others need to add skrill themselves - although technically could be tied into their local currency national banks? (that'd be more liquid, right?)

**Comment by pazza83** [comment]
@sqrrm yes you are correct it is buy only.

- Maker as Buy BTC
- Taker as Buy BTC

>How would this work? User sets up a Skrill account, won't be able to create offers but it would match to sepa/faster payment/wire/national bank options while looking to take offers?

Yes that would be correct.

> Does the information for all those methods fit with the skrill side of things or is something extra needed?

Yes it does. Buyer (Skrill account holder) would just need to enter sellers payment details when making payment using Skrill. Seller would receive funds from Skrill's account.

@Conza88 
> Is it just opening access to national/local banks? Or would others need to add skrill themselves - although technically could be tied into their local currency national banks? (that'd be more liquid, right?)

It would open up access to users wanting to buy in any of the following currencies irrespective of their location:

AUD

- BDT
- BRL
- EUR
- GBP
- IDR
- INR
- KES
- LKR
- MXN
- MYR
- NPR
- PHP
- PKP
- PLN
- THB
- USD
- VDN

**Comment by sqrrm** [comment]
It sounds reasonable to add this. I haven't looked at the details regarding checks to not allow selling for some account types but I suspect it's won't be hard to implement.

**Comment by pazza83** [comment]
Skrill has been scored for inclusion to Bisq:


| Essential   | Desirable | Definite No’s |
| ------------------- | ------------------- | ------------------- |
| **Very low risk of chargeback**  | ~~No risk of chargeback~~  | ~~< very low risk of chargeback~~  |
| ~~Way to verify the sender in the received payment~~  | **Way to verify the sender in the received payment and ability to enter a reference**  | ~~No way to verify the sender in the received payment~~  |
| **Trade time less than one week**  | ~~Instant payment~~  | ~~Trade time more than one week~~  |
| ~~Singular Fiat currency~~  | **Multi-currency**  | ~~Not a payment method for fiat currency~~  |
| ~~Significant user base~~  |  **Large user base**  | ~~No significant user base~~  |
| ~~High usability~~  | **High usability and great user experience**  | ~~< high amount usability~~  |
| ~~No KYC required for sending and receiving payments~~ | **No KYC required for sending and receiving payments, allows users to trade with upmost privacy. Minimal identifying information as possible (no names, email, phone etc required)** | ~~Some KYC required (proof of address, ID, selfie) for sending and receiving payments~~  |
| **Low risk of scam attempts**  | ~~Very low risk of scam attempts~~  | ~~< low risk of scam attempts~~ |
| ~~Traders can provide evidence of payment / receipt~~  | **Traders can provide evidence of payment / receipt and Verification of payment can be made using PageSigner or similar** | ~~Traders will be unable to provide evidence of payment / receipt~~  |
| ~~Minimum limit at least equal to at least account limits protocols~~  | **No minimum limits**  | ~~Minimum limit not able to achieve account limits protocols~~  |
| ~~Maximum limits equal to at least 0.01 BTC~~  | **Large payment limits up to 2 BTC**  | ~~Maximum limit is less than 0.01 BTC~~  |
| ~~Likely to increase liquidity~~  | **Likely to increase liquidity and open markets for different countries and currencies**  | ~~Likely to decrease liquidity~~  |
| **Low risk of mediation**  | ~~Very low risk of mediation~~  | ~~< low risk of mediation~~  |
| **Low risk for traders from government agencies**  | ~~No risk for traders from government agencies~~ | ~~< low risk for traders from government agencies~~  |
| **Fees should not be a barrier to trading**  | ~~No fees for transactions~~  | ~~Fees will be a barrier to trading~~  |
| **Only minor changes needed to trade protocol**  | ~~No changes needed to trade protocol~~  | ~~> Minor changes needed to trade protocol~~  |

Score: 25/32
Percentage: 78.13%

**Comment by pazza83** [comment]
### Skrill - Proposed National Currency Payment Method 

I am proposing that Skrill be added as a national currency payment method.

@refund-agent2, @huey735, @leo816 please let me know if you have any questions as to how Skrill will be mediated / arbitrated. If you have any objection please let me know by 31 May 2021.

---

Date proposal being made: 02.05.21
Date feedback to be provided by: 31.05.21

Name of payment method: Skrill 

Required fields (eg)

- Email

Currencies accepted

- AUD
- BDT
- BRL
- EUR
- GBP
- IDR
- INR
- KES
- LKR
- MXN
- MYR
- NPR
- PHP
- PKP
- PLN
- THB
- USD
- VDN

- Maximum trade duration: 24 hours

Signing required: No

Bisq Limits: eg 0.25 BTC

Trade window: 6 days

Payment method limits: Please create as a 'buy-only' payment method (see notes above). Skrill accounts would be able to buy from any other following offers:

- SEPA (EUR)
- Faster Payments (GBP)
- Wire Transfer US (USD)
- National Bank Transfer (in any of the currencies above)

Any relevant information for users: None

Screenshot of expected GUI: TBC

**Comment by pazza83** [comment]
This payment method is a more complicated to add then the others.

User that sets up a Skrill account will only be able make offers to buy BTC AND take offers to sell BTC.

Make offers to buy BTC would be able to be taken by any of the following payment methods:

- SEPA (EUR)
- Faster Payments (GBP)
- Wire Transfer US (USD)
- National Bank Transfer (in any of the currencies above)

Take offers to sell BTC would be able to be taken by any of the following payment methods:

- SEPA (EUR)
- Faster Payments (GBP)
- Wire Transfer US (USD)
- National Bank Transfer (in any of the currencies above)

Limits, trade windows will be determined by the sellers of BTC not the buyer of BTC.

Here is the UX design:

![Payment method - Skrill](https://user-images.githubusercontent.com/66367547/117580584-7f574f00-b0f0-11eb-8e70-9c1e9056bc25.png)

**Comment by pazza83** [comment]
NB: Design above should replace 'Email' with 'Account owner full name'

**Comment by pazza83** [comment]
## Information required for Bisq software strings:

Hi @jmacxx here is the information for the strings. 

Let me know if you need anything else:

### Special info about any of the inputs required for creating the payment account:

Please note Skrill is a payment method that allows users to buy bitcoin in multiple fiat currencies. It is not possible to use sell to sell bitcoin.

It is, therefore, possible to use a Skrill account to:

- Make offers to buy BTC that can be taken by a BTC seller
- Take offers to buy BTC from a BTC seller

Please ensure to include your full name that matches your full name in Skill. This will ensure that when you send funds they show from being from yourself.

### Special items to be aware about when sending payments (for BTC Buyers):

Please send payment only to the payment details of the BTC Seller as shown in Bisq. 

Please send payment only to the payment in the currency that matches the market that the offer took place on.

It is recommend to use your full name that matches your full name in Skill in the payment description / reason for payment field. This will ensure that when you send funds they show from being from yourself.

### Special items to be aware about when receiving payments (for BTC Sellers):

BTC Sellers should expect to receive payment via a  third party (Skrill partner bank). This payment should be in the currency for the exact amount traded. The payee's name should be shown within the payment description or payee information. This allows you to identify the payment as being from the BTC buyer.
